<p>There is no good reason to declare a field "public" and "static" without also declaring it "final". Most of the time this is a kludge to share a state among several objects. But with this approach, any object can do whatever it wants with the shared state, such as setting it to <code>null</code>.</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class Greeter {
public static var foo:Foo = new Foo(...);
...
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class Greeter {
public static const FOO:Foo = new Foo(...);
...
}
</pre>